### PR TITLE
some cleanup of file.c

### PR DESCRIPTION
**Pull request by adizaimi**
Cleanup file.c a bit and more:
* use length returned from snprintf rather than running strlen uselessly
* use sizeof(char array[]) - 1 instead of strlen(array) in several cases
* strlen(name) outside of loop
* shortcut with continue while loops rather than having three indentation levels
* more tables in test to check for leaks etc.

### CODE REVIEW

File: bdb/file.c [hunk 1]
```diff
@@ -7662,159 +7664,156 @@ static int bdb_process_unused_files(bdb_state_type *bdb_state, tran_type *tran,
...
+                    *p_buf_end = p_buf + sizeof(invers);
+            struct bdb_file_version_num_type p_file_version_num_type;
+
+            bdb_file_version_num_get(&p_file_version_num_type, p_buf,
+                                     p_buf_end);
+
+            file_version = p_file_version_num_type.version_num;
+        }
+
+        if (file_version)
```
Comment (ldennis) [inline bdb/file.c]: I think this is wrong. Schema change is not deleting files now. Should be `if (!file_version)`

File: bdb/file.c [hunk 2]
```diff
@@ -7662,159 +7664,156 @@ static int bdb_process_unused_files(bdb_state_type *bdb_state, tran_type *tran,
...
+        /* try to find the file version amongst the active data files */
+        for (i = 0; i < bdb_state->numdtafiles; ++i) {
+            if (bdb_state->bdbtype == BDBTYPE_QUEUEDB) {
+                rc = bdb_get_file_version_qdb(bdb_state, tran, &version_num,
+                                              bdberr);
+            } else {
+                rc = bdb_get_file_version_data(bdb_state, tran, i, &version_num,
+                                               bdberr);
+            }
+            if (rc == 0 && version_num == file_version) {
```
Comment (ldennis) [inline bdb/file.c]: I just figured out that this is also wrong.. We dont want to break if rc == 0..

Comment (adizaimi) [reply bdb/file.c]: That would be behavior different from original. It seems that we declare found_in_llmeta = 1 if rc==0 and if version retrieved matched.  I did not write the original but it looks correct.

File: bdb/file.c (hunk 1 above)
Comment (adizaimi) [reply bdb/file.c]: thanks for catching. we want to continue if !file_version.

File: bdb/file.c (hunk 2 above)
Comment (ldennis) [reply bdb/file.c]: I am saying the case where if rc == 0 but version_num != file_version, your patch will break out of the for loop but we should continue instead.

Comment (ldennis) [reply bdb/file.c]: try create a table with blob, new code will mistakenly delete all blob files.

File: bdb/file.c [hunk 3]
```diff
@@ -7662,159 +7664,156 @@ static int bdb_process_unused_files(bdb_state_type *bdb_state, tran_type *tran,
...
-                for (i = 0; i < bdb_state->numdtafiles; ++i) {
-                    if (bdb_state->bdbtype == BDBTYPE_QUEUEDB) {
-                        rc = bdb_get_file_version_qdb(bdb_state, tran,
-                                                      &version_num, bdberr);
-                    } else {
-                        rc = bdb_get_file_version_data(bdb_state, tran, i,
-                                                       &version_num, bdberr);
-                    }
-                    if (rc == 0) {
-                        if (version_num == file_version) {
```
Comment (ldennis) [inline bdb/file.c]: FYI: here is the original logic

File: bdb/file.c (hunk 2 above)
Comment (ldennis) [reply bdb/file.c]: scindex catches it occasionally because of missing files. But I think we should really have a dedicated test for file deletion.

Comment (adizaimi) [reply bdb/file.c]: You are a genius for catching this!

**Pull request merged**